qtt/instrument_drivers/virtualAwg/awgs/ZurichInstrumentsHDAWG8.py

```diff
@@ -66,94 +74,37 @@ def retrieve_running_mode(self):
         raise NotImplementedError
 
     def update_sampling_rate(self, sampling_rate):
-        raise NotImplementedError
+        for key, value in self.__sampling_rate_map.items():
+            if sampling_rate == value:
+                self.__awg.set('awgs_{}_time'.format(self.__awg_number), key)
+                return
+        raise ValueError('Sampling rate {} not in available a list of available values: {}'.format(
+            sampling_rate, self.__sampling_rate_map))
 
     def retrieve_sampling_rate(self):
-        #ziDAQ('setInt', '/dev8044/awgs/0/time', 0);
-        return 2.4e9
+        sample_rate = self.__awg.get('awgs_{}_time'.format(self.__awg_number))
```

Comment:
If consistent with the name of the function, it should be `sampling_rate` instead of `sample_rate`. Also, it is not the sampling rate, but rather an index. So maybe `sampling_rate_index` or `sampling_rate_key`